Validation/HcalDigis/src/HcalDigisValidation.cc

```diff
@@ -528,17 +532,18 @@ void HcalDigisValidation::analyze(const edm::Event& iEvent, const edm::EventSetu
 
    for (HcalTrigPrimDigiCollection::const_iterator itr = dataTPs->begin(); itr != dataTPs->end(); ++itr) {
      int ieta  = itr->id().ieta();
-     int iphi  = itr->id().iphi();
 
      HcalSubdetector subdet = (HcalSubdetector) 0;
-     if      ( abs(ieta) <= 16 ) subdet = HcalSubdetector::HcalBarrel ;
-     else if ( abs(ieta) <= 28 ) subdet = HcalSubdetector::HcalEndcap ;
-     else if ( abs(ieta) <= 40 ) subdet = HcalSubdetector::HcalForward;
+     if      ( abs(ieta) <= 16 )
+        subdet = HcalSubdetector::HcalBarrel ;
+     else if ( abs(ieta) < tp_geometry->firstHFTower(itr->id().version()) ) 
+        subdet = HcalSubdetector::HcalEndcap ;
+     else if ( abs(ieta) <= 42 )
```

Comment:
to me this last check also seems to be redundant (just "else" would be good 
enough), but doesn't hurt. 
we actually had previously <= 32 (not really 40), now we can have <= 42 
(for 1x1 TPs).


On Thu, 17 Mar 2016, David Lange wrote:

> 
> In Validation/HcalDigis/src/HcalDigisValidation.cc:
> 
> > 
> >       HcalSubdetector subdet = (HcalSubdetector) 0;
> > -     if      ( abs(ieta) <= 16 ) subdet = HcalSubdetector::HcalBarrel ;
> > -     else if ( abs(ieta) <= 28 ) subdet = HcalSubdetector::HcalEndcap ;
> > -     else if ( abs(ieta) <= 40 ) subdet = HcalSubdetector::HcalForward;
> > +     if      ( abs(ieta) <= 16 )
> > +        subdet = HcalSubdetector::HcalBarrel ;
> > +     else if ( abs(ieta) < tp_geometry->firstHFTower(itr->id().version()) ) 
> > +        subdet = HcalSubdetector::HcalEndcap ;
> > +     else if ( abs(ieta) <= 42 )
> 
> @abdoulline -I don't follow the last if statement. 40 changes to 42 - but why is it needed
> at all?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on
> GitHub[AEx02iQlsw1Cvqs3XCMma6JiP5NG5Pwnks5puVaEgaJpZM4Hc6h-.gif]
> 
> 
>